### PR TITLE
Tixxdz/fleetctl improve unit lookup and wait

### PR DESCRIPTION
**Pull request by tixxdz**

### CODE REVIEW

**Comment by dongsupark** [comment]
This tree looks good. But when getting into the detail, it's a little hard to follow each commit. Would it be better to reduce the number of commits, for example, by squashing relevant commits around GetUnitInstanceInfo()?

**Comment by tixxdz** [comment]
@dongsupark you are right! and I'm trying to do that, yes the branch got long after the review from Jon, you can check the relevant PR here: https://github.com/coreos/fleet/pull/1433 So I did handle all Jon comments. I force pushed that branch so commits were squashed! however I opened this PR here in our branch to track a CI functional test that was failing, I wanted to test it also before merging that PR... still in progress. thank you!
